packages/p2p/src/pages/my-ads/ad-summary/ad-summary.tsx

```diff
@@ -0,0 +1,118 @@
+            : formatMoney(local_currency_config.currency, Number(offer_amount) * Number(price_rate), true);
+    }
+
+    if (offer_amount) {
+        const components = [
+            <Text key={0} className='ad-summary' weight='bold' size='xs' color='status-info-blue' />,
+            <Text key={1} className='ad-summary' size='xs' color='status-info-blue' />,
+        ];
+        const values = { target_amount: display_offer_amount, target_currency: currency, ad_text };
+        if (price_rate) {
+            Object.assign(values, {
+                local_amount: display_total,
+                local_currency: local_currency_config.currency,
+                price_rate: removeTrailingZeros(
+                    formatMoney(local_currency_config.currency, display_price_rate, true, 6)
+                ),
+            });
+
+            if (type === buy_sell.BUY) {
+                return (
+                    <Text className='ad-summary' size='xs' color='less-prominent'>
+                        <Localize
+                            i18n_default_text="You're {{ ad_text }} an ad to buy <0>{{ target_amount }} {{ target_currency }}</0> for <0>{{ local_amount }} {{ local_currency }}</0> <1>({{ price_rate }} {{local_currency}}/{{ target_currency }})</1>"
+                            components={components}
+                            values={values}
+                        />
+                    </Text>
+                );
+            }
+
+            return (
+                <Text className='ad-summary' size='xs' color='less-prominent'>
+                    <Localize
+                        i18n_default_text="You're {{ ad_text }} an ad to sell <0>{{ target_amount }} {{ target_currency }}</0> for <0>{{ local_amount }} {{ local_currency }}</0> <1>({{ price_rate }} {{local_currency}}/{{ target_currency }})</1>"
+                        components={components}
+                        values={values}
+                    />
+                </Text>
+            );
+        }
+
+        if (type === buy_sell.BUY) {
+            return (
+                <Text className='ad-summary' size='xs' color='less-prominent'>
+                    <Localize
+                        i18n_default_text="You're {{ ad_text }} an ad to buy <0>{{ target_amount }} {{ target_currency }}</0>..."
+                        components={components}
+                        values={values}
+                    />
+                </Text>
+            );
+        }
+
+        return (
+            <Text className='ad-summary' size='xs' color='less-prominent'>
+                <Localize
+                    i18n_default_text="You're {{ ad_text }} an ad to sell <0>{{ target_amount }} {{ target_currency }}</0>..."
+                    components={components}
+                    values={values}
+                />
+            </Text>
+        );
+    }
+
+    return type === buy_sell.BUY ? (
```

Comment:
Can we move this condition inside the component instead. Since the same text component is called twice